tests/controller_tests/test_retrieve.py

```diff
@@ -0,0 +1,19 @@
+from sqlalchemy import select
+
+from core.const import EXCEL_MIMETYPE
+from core.db import db
+from core.db.entities import Submission
+
+
+def test_retrieve_invalid_UUID(seeded_test_client):
+    invalid_UUID = "123456ab-0cd7-89ef-1234-5g678h91ijk2"
+    response = seeded_test_client.get(f"/retrieve?submission_UUID={invalid_UUID}&attachment=true")
+    assert response.status_code == 404
+
+
+def test_retrieve(seeded_test_client):
+    submission_UUID = db.session.scalar(select(Submission.id))
+    response = seeded_test_client.get(f"/retrieve?submission_UUID={submission_UUID}&attachment=false")
+    assert response.status_code == 200
+    assert response.headers["content-type"] == "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+    assert response.content_type == EXCEL_MIMETYPE
```

Comment:
Looks like these should be expanded to assert the rest of the response that is in the schema